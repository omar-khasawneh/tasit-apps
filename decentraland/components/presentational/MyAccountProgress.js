import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import { StyleSheet, Text, View } from "react-native";
import React from "react";
import Colors from "@constants/Colors";
import ProgressBar from "react-native-progress/Bar";
import PropTypes from "prop-types";

export default function MyAccountProgress({ progress }) {
  return (
    <View style={styles.container}>
      <ProgressBar
        progress={progress}
        color={Colors.loadingColor}
        borderWidth={0}
        unfilledColor={Colors.textColor}
        height={responsiveHeight(1)}
        width={responsiveWidth(101)}
        style={styles.progress}
      />
      <View style={styles.progressTextContainer}>
        <Text style={styles.progressText}>
          {Math.round(progress * 100)}% complete
        </Text>
      </View>
    </View>
  );
}

MyAccountProgress.propTypes = {
  progress: PropTypes.number,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: Colors.backgroundColor,
    alignItems: "center",
    justifyContent: "flex-start",
  },
  progress: {
    marginTop: responsiveHeight(10),
  },
  progressTextContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  progressText: {
    fontWeight: "800",
  },
});
