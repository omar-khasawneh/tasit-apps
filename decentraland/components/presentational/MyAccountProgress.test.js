import React from "react";
import { shallow } from "enzyme";
import MyAccountProgress from "./MyAccountProgress";

describe("MyAccountProgress", () => {
  it("renders the component", async () => {
    const progress = 0.5;
    expect(
      shallow(<MyAccountProgress progress={progress} />)
    ).toMatchSnapshot();
  });
});
