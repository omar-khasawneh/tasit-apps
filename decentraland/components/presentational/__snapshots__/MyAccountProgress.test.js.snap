// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccountProgress renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={4}
    borderWidth={0}
    color="#2196f3"
    height={13.34}
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    progress={0.5}
    style={
      Object {
        "marginTop": 133.4,
      }
    }
    unfilledColor="gray"
    useNativeDriver={false}
    width={757.5}
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "fontWeight": "800",
        }
      }
    >
      50
      % complete
    </Component>
  </Component>
</Component>
`;
